### DISCUSSION TITLE
How to set up separate local repository for a branch?

### DISCUSSION THREAD

**Discussion by Pandamander** [community/community#23235] [New to GitHub]
<p>Hi everyone, I’m using Github Desktop and my team has set up a new branch. I’d like to set up a separate local repository for this branch: one folder on my mac for master, and another folder for branch.</p>
<p>Is there a way to do this? I keep finding tutorials for how to set up a branch, but not how to make a second local repo.</p>
<p>Thank you!</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>I think <code>git worktree</code> is what you want: It lets you have multiple working directories (for example using different branches) using one local repository, without duplicating all the other repository data. I don’t know if Github Desktop can manage worktrees.</p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://git-scm.com/favicon.ico" class="site-icon" width="16" height="16">
      <a href="https://git-scm.com/docs/git-worktree" target="_blank" rel="noopener">git-scm.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="16" height="16">

<h3><a href="https://git-scm.com/docs/git-worktree" target="_blank" rel="noopener">Git - git-worktree Documentation</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
